src/main/scala/ylabs/messaging/Client.scala

```diff
@@ -97,18 +108,26 @@ class Client extends FSM[State, Context] {
     case Event(Messages.RegisterEventListener(actor), ctx) ⇒
       stay using ctx.copy(eventListeners = ctx.eventListeners + actor)
 
-    case Event(msg: Messages.ListenerEvent, ctx) ⇒
+    case Event(msg: Messages.ListenerEvent, ctx @ Context(Some(connection), chats, _)) ⇒
```

Comment:
if you replace the _ in the pattern match with `listeners` then you don't need `ctx @`